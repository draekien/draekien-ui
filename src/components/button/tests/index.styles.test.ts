import {
  buttonCss,
  linkButtonCss,
  iconWrapperCss,
  ButtonCssProps,
} from '../index.styles';

describe('buttonCss function', () => {
  test('returns primary colors when variant is primary', () => {
    const props: ButtonCssProps = {
      variant: 'primary',
    };

    const css = buttonCss(props);

    expect(css).toMatchSnapshot();
  });

  test('returns secondary css when variant is secondary', () => {
    const props: ButtonCssProps = {
      variant: 'secondary',
    };

    const css = buttonCss(props);

    expect(css).toMatchSnapshot();
  });

  test('contains disabled css when disabled', () => {
    const props: ButtonCssProps = {
      variant: 'primary',
      disabled: true,
    };

    const css = buttonCss(props);

    expect(css).toMatchSnapshot();
  });

  test('returns expected css when active', () => {
    const props: ButtonCssProps = {
      variant: 'primary',
      active: true,
    };

    const css = buttonCss(props);

    expect(css).toMatchSnapshot();
  });

  test('returns expected css when fullwidth', () => {
    const props: ButtonCssProps = {
      variant: 'primary',
      active: true,
      fullWidth: true,
    };

    const css = buttonCss(props);

    expect(css).toMatchSnapshot();
  });

  test('returns expected css when variant is text', () => {
    const props: ButtonCssProps = {
      variant: 'text',
      fullWidth: true,
    };

    const css = buttonCss(props);

    expect(css).toMatchSnapshot();
  });
});

describe('linkButtonCss function', () => {
  test('returns expected css when not active', () => {
    expect(linkButtonCss(false)).toMatchSnapshot();
  });

  test('returns expected css when active', () => {
    expect(linkButtonCss(true)).toMatchSnapshot();
  });
});

describe('iconWrapperCss function', () => {
  test('returns expected css when has text and position left', () => {
    expect(iconWrapperCss('left', true)).toMatchSnapshot();
  });

  test('returns expected css when has text and position left', () => {
    expect(iconWrapperCss('left', false)).toMatchSnapshot();
  });

  test('returns expected css when has text and position right', () => {
    expect(iconWrapperCss('right', true)).toMatchSnapshot();
  });

  test('returns expected css when no text', () => {
    expect(iconWrapperCss('right', false)).toMatchSnapshot();
  });
});
