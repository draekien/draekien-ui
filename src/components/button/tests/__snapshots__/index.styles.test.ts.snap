// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`buttonCss function contains disabled css when disabled 1`] = `
Object {
  ":hover::before": Object {
    "backgroundImage": "none",
  },
  "alignItems": "center",
  "backgroundColor": "muted",
  "backgroundImage": "none",
  "border": "0.0625rem solid",
  "borderColor": "muted",
  "borderRadius": "md",
  "color": "white",
  "cursor": "not-allowed",
  "display": "inline-flex",
  "height": "2rem",
  "justifyContent": "center",
  "px": "md",
  "py": "xs",
  "textAlign": "center",
  "textDecoration": "none",
  "textTransform": "none",
  "transition": "all 300ms",
  "variant": "text.subtitle",
}
`;

exports[`buttonCss function returns expected css when active 1`] = `
Object {
  ":active": Object {
    "backgroundColor": "primary-active",
    "borderColor": "primary-active",
    "span": Object {
      "color": "white",
    },
  },
  ":hover": Object {
    "backgroundColor": "primary-hover",
    "borderColor": "primary-hover",
    "span": Object {
      "color": "white",
    },
  },
  "alignItems": "center",
  "backgroundColor": "primary-active",
  "border": "0.0625rem solid",
  "borderColor": "primary-active",
  "borderRadius": "md",
  "color": "white",
  "cursor": "pointer",
  "display": "inline-flex",
  "height": "2rem",
  "justifyContent": "center",
  "px": "md",
  "py": "xs",
  "span": Object {
    "color": "white",
  },
  "textAlign": "center",
  "textDecoration": "none",
  "textTransform": "none",
  "transition": "all 300ms",
  "variant": "text.subtitle",
}
`;

exports[`buttonCss function returns expected css when fullwidth 1`] = `
Object {
  ":active": Object {
    "backgroundColor": "primary-active",
    "borderColor": "primary-active",
    "span": Object {
      "color": "white",
    },
  },
  ":hover": Object {
    "backgroundColor": "primary-hover",
    "borderColor": "primary-hover",
    "span": Object {
      "color": "white",
    },
  },
  "alignItems": "center",
  "backgroundColor": "primary-active",
  "border": "0.0625rem solid",
  "borderColor": "primary-active",
  "borderRadius": "md",
  "color": "white",
  "cursor": "pointer",
  "display": "flex",
  "height": "2rem",
  "justifyContent": "center",
  "px": "md",
  "py": "xs",
  "span": Object {
    "color": "white",
  },
  "textAlign": "center",
  "textDecoration": "none",
  "textTransform": "none",
  "transition": "all 300ms",
  "variant": "text.subtitle",
  "width": "100%",
}
`;

exports[`buttonCss function returns expected css when variant is text 1`] = `
Object {
  ":active": Object {
    "backgroundColor": "transparent",
    "borderColor": "transparent",
    "span": Object {
      "color": "primary-active",
    },
  },
  ":hover": Object {
    "backgroundColor": "transparent",
    "borderColor": "transparent",
    "span": Object {
      "color": "primary-hover",
    },
  },
  "alignItems": "center",
  "backgroundColor": "transparent",
  "border": "0.0625rem solid",
  "borderColor": "transparent",
  "borderRadius": "md",
  "color": "primary",
  "cursor": "pointer",
  "display": "flex",
  "height": "2rem",
  "justifyContent": "center",
  "p": "0 !important",
  "px": "md",
  "py": "xs",
  "span": Object {
    "color": "primary",
  },
  "textAlign": "center",
  "textDecoration": "none",
  "textTransform": "none",
  "transition": "all 300ms",
  "variant": "text.subtitle",
  "width": "100%",
}
`;

exports[`buttonCss function returns primary colors when variant is primary 1`] = `
Object {
  ":active": Object {
    "backgroundColor": "primary-active",
    "borderColor": "primary-active",
    "span": Object {
      "color": "white",
    },
  },
  ":hover": Object {
    "backgroundColor": "primary-hover",
    "borderColor": "primary-hover",
    "span": Object {
      "color": "white",
    },
  },
  "alignItems": "center",
  "backgroundColor": "primary",
  "border": "0.0625rem solid",
  "borderColor": "primary",
  "borderRadius": "md",
  "color": "white",
  "cursor": "pointer",
  "display": "inline-flex",
  "height": "2rem",
  "justifyContent": "center",
  "px": "md",
  "py": "xs",
  "span": Object {
    "color": "white",
  },
  "textAlign": "center",
  "textDecoration": "none",
  "textTransform": "none",
  "transition": "all 300ms",
  "variant": "text.subtitle",
}
`;

exports[`buttonCss function returns secondary css when variant is secondary 1`] = `
Object {
  ":active": Object {
    "backgroundColor": "secondary-active",
    "borderColor": "secondary-active",
    "span": Object {
      "color": "white",
    },
  },
  ":hover": Object {
    "backgroundColor": "secondary-hover",
    "borderColor": "secondary-hover",
    "span": Object {
      "color": "white",
    },
  },
  "alignItems": "center",
  "backgroundColor": "secondary",
  "border": "0.0625rem solid",
  "borderColor": "secondary",
  "borderRadius": "md",
  "color": "white",
  "cursor": "pointer",
  "display": "inline-flex",
  "height": "2rem",
  "justifyContent": "center",
  "px": "md",
  "py": "xs",
  "span": Object {
    "color": "white",
  },
  "textAlign": "center",
  "textDecoration": "none",
  "textTransform": "none",
  "transition": "all 300ms",
  "variant": "text.subtitle",
}
`;

exports[`iconWrapperCss function returns expected css when has text and position left 1`] = `
Object {
  "display": "inline-flex",
  "marginRight": "xxs",
  "textDecoration": "none",
}
`;

exports[`iconWrapperCss function returns expected css when has text and position left 2`] = `
Object {
  "display": "inline-flex",
  "i": Object {
    "fontSize": "1.5rem",
  },
  "textDecoration": "none",
}
`;

exports[`iconWrapperCss function returns expected css when has text and position right 1`] = `
Object {
  "display": "inline-flex",
  "marginLeft": "xxs",
  "textDecoration": "none",
}
`;

exports[`iconWrapperCss function returns expected css when no text 1`] = `
Object {
  "display": "inline-flex",
  "i": Object {
    "fontSize": "1.5rem",
  },
  "textDecoration": "none",
}
`;

exports[`linkButtonCss function returns expected css when active 1`] = `
Object {
  ":active": Object {
    "span": Object {
      "color": "primary-active",
      "textDecoration": "underline",
      "transition": "all 300ms",
    },
  },
  ":disabled": Object {
    "cursor": "not-allowed",
    "span": Object {
      "color": "muted",
      "textDecoration": "none",
      "transition": "all 300ms",
    },
  },
  ":hover": Object {
    "span": Object {
      "color": "primary-hover",
      "textDecoration": "underline",
      "transition": "all 300ms",
    },
  },
  "alignItems": "center",
  "backgroundColor": "transparent",
  "border": 0,
  "color": "primary",
  "cursor": "pointer",
  "display": "inline-flex",
  "height": "auto",
  "justifyContent": "center",
  "padding": 0,
  "span": Object {
    "color": "primary-active",
    "textDecoration": "underline",
    "transition": "all 300ms",
  },
  "transition": "all 300ms",
  "variant": "text.subtitle",
  "width": "auto",
}
`;

exports[`linkButtonCss function returns expected css when not active 1`] = `
Object {
  ":active": Object {
    "span": Object {
      "color": "primary-active",
      "textDecoration": "underline",
      "transition": "all 300ms",
    },
  },
  ":disabled": Object {
    "cursor": "not-allowed",
    "span": Object {
      "color": "muted",
      "textDecoration": "none",
      "transition": "all 300ms",
    },
  },
  ":hover": Object {
    "span": Object {
      "color": "primary-hover",
      "textDecoration": "underline",
      "transition": "all 300ms",
    },
  },
  "alignItems": "center",
  "backgroundColor": "transparent",
  "border": 0,
  "color": "primary",
  "cursor": "pointer",
  "display": "inline-flex",
  "height": "auto",
  "justifyContent": "center",
  "padding": 0,
  "span": Object {
    "color": "primary",
  },
  "transition": "all 300ms",
  "variant": "text.subtitle",
  "width": "auto",
}
`;
